### PR TITLE
add some documentation about using sccache with this image

### PR DESCRIPTION
**Pull request by cbeck88**
as requested (https://github.com/clux/muslrust/pull/144#pullrequestreview-2072561465), here's some docu about using sccache

let me know if this looks okay to you. thanks

### CODE REVIEW

File: SCCACHE.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+### Post-build command
+
+As described in the main [`README.md`](./README.md), on linux the build is going to run as root in the `muslrust` image and so any files it produces will be owned by root, if they are mounted into the container.
+For several reasons that can become annoying, and a quick `chown` fixes it.
+
+Here we're adding a `POST_BUILD_COMMAND` that changes ownership not only for the `target` directory, but also the cargo registry and sccache directories. This is because the github `actions/cache` action will fail
+to save and restore files owned by root.
+
+On a mac, docker works differently, so if you are using the example command there, the files won't actually be owned by root, and also the `chown` command will be very slow. So on mac it is better to either skip
+the `POST_BUILD_CMD`, or you could modify it so that it actually tests if we are root before doing the `chown`.
```
Comment (clux) [inline SCCACHE.md:128]: huh, TIL. guess that's what i get for never bothering to test mac.

**Review by clux — APPROVED** [review]
wow, great and cohesive e2e guide. thank you very much. i really appreciate it.

this largely tracks with my expectations, so excited to try this out myself and see. happy to send this out as-is.

**Comment by cbeck88** [comment]
actually there's some more stuff i was just reading about, apparently you can try to make sccache talk directly the github actions cache api instead of using a local directory to store and retrieve artifacts: 

https://github.com/mozilla/sccache/blob/main/docs/GHA.md

then there are various improvements people made to avoid getting rate limited...

https://github.com/mozilla/sccache/pull/2142
https://github.com/Mozilla-Actions/sccache-action/issues/50

i haven't used any of this before. probably those who need it will find it on their own, i'm not having any problems with the setup i described here anyways

**Comment by clux** [comment]
> actually there's some more stuff

yeah, all of that also sounds useful, i also have not tried these.
if anyone wants to contribute useful links to working setups / pointers or such in the future, that would be welcome.

in the meantime, merging this. thanks again!

**Pull request merged**